websocket/websocket-server.go

```diff
@@ -0,0 +1,335 @@
+	// In this case, it's set to always return true, meaning any origin is accepted.
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
+
+// RunWebsocketServer initializes and runs the WebSocket server.
+func RunWebsocketServer(rt *runtime.Runtime) {
+
+	router := mux.NewRouter()
+
+	// Set up CORS handling
+	headers := handlers.AllowedHeaders([]string{"*"})
+	origins := handlers.AllowedOrigins([]string{"*"}) // Allow requests from any origin
+
+	// Add CORS middleware
+	router.Use(handlers.CORS(headers, origins))
+
+	router.HandleFunc("/ws", func(w http.ResponseWriter, r *http.Request) {
+		ClientSocket(w, r, rt)
+	})
+	fmt.Println("Server is running on port 8085")
+
+	// Start your server
+	go func() {
+		err := http.ListenAndServe(":8085", router)
+		if err != nil {
+			log.Println(err)
+		}
+	}()
+}
+
+// ClientSocket handles incoming WebSocket connections from clients.
+func ClientSocket(w http.ResponseWriter, r *http.Request, rt *runtime.Runtime) {
+
+	Conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Print("upgrade:", err)
+		return
+	}
+	defer func(Conn *websocket.Conn) {
+		err := Conn.Close()
+		if err != nil {
+			log.Println(err)
+		}
+	}(Conn)
+
+	rt.Service()
+	service := rt.Service()
+
+	for {
+		messageType, requestByte, err := Conn.ReadMessage()
+
+		if err != nil {
+			log.Println("ERROR read: ", err.Error())
+			break
+		}
+
+		requestJson := make(map[string]interface{})
+		err = json.Unmarshal(requestByte, &requestJson)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		switch requestJson[RequestParamRequestType] {
+		case RequestTypeChatRoom:
+
+			if _, ok := requestJson[RequestParamRoomId]; ok {
+				if !ok {
+					Response := make(map[string]interface{})
+					Response[RequestParamRequestType] = RequestTypeChatRoom
+					Response[RequestParamMessage] = "required " + RequestParamRoomId
+					responseBytes, _ := json.Marshal(Response)
+					WriteOnSocket(responseBytes, messageType, Conn)
+					break
+				}
+			}
+
+			if _, ok := requestJson[RequestParamSenderId]; ok {
+				if !ok {
+					Response := make(map[string]interface{})
+					Response[RequestParamRequestType] = RequestTypeChatRoom
+					Response[RequestParamMessage] = "required " + RequestParamSenderId
+					responseBytes, _ := json.Marshal(Response)
+					WriteOnSocket(responseBytes, messageType, Conn)
+					break
+				}
+			}
+
+			roomId, ok := requestJson[RequestParamRoomId].(string)
+
+			if !ok {
+				Response := make(map[string]interface{})
+				Response[RequestParamRequestType] = RequestTypeChatRoom
+				Response[RequestParamMessage] = "required string type " + RequestParamRoomId
+				responseBytes, _ := json.Marshal(Response)
+				WriteOnSocket(responseBytes, messageType, Conn)
+				break
+			}
+
+			senderId, ok := requestJson[RequestParamSenderId].(string)
+			if !ok {
+				Response := make(map[string]interface{})
```

Comment:
avoid creating a same variable in each case declare this on top of switch

Reply:
Done